app/services/metrics/review_turnaround/per_department.rb

```diff
@@ -0,0 +1,44 @@
+module Metrics
+  module ReviewTurnaround
+    class PerDepartment < Metrics::Base
+      def initialize(interval = nil)
+        @interval = interval
+      end
+
+      def call
+        process
+      end
+
+      private
+
+      def process
+        department_metrics_values.each do |department_id, metrics_values|
+          turnaround = calculate_turnaround(metrics_values)
+          create_or_update_metric(department_id, Department.to_s, metric_interval,
+                                  turnaround, :review_turnaround)
+        end
+      end
+
+      def department_metrics_values
+        Metric.joins(project: :department)
+              .where(name: :review_turnaround)
+              .pluck('departments.id', :value)
+              .each_with_object({}) do |(id, value), hash|
+                (hash[id] ||= []) << value
+                hash
+              end
+      end
+
+      def calculate_turnaround(metrics_values)
+        metrics_values_size = metrics_values.size
+        return metrics_values.first if metrics_values_size == 1
```

Comment:
We can remove this line and it won't change anything right? If `metrics_values_size` is 1, then `metrics_values.sum / metrics_values_size` will be `metrics_values.first`

Reply:
I am not sure about `/` meaning between 2 integers. If it happens to be an integer division the remainder would be lost.
I agree with you but would use

```
metrics_values.sum / metrics_values_size.to_f
```

to ensure it's not an integer division

Reply:
@mrubi-rootstrap in this method we always receive the metric value, which is a decimal value, so never will be a integer value.

Reply:
All right. Since that is something that depends on a parameter received I would still ensure the `metrics_values_size.to_f` to be a non integer Numeric or I would add a test to flag an error if the parameter received happens to be of a different type though. Just to make sure